File: go/src/other/dp/BurstingBalloon.go
package dp

func MaxCoins(nums []int) int {
	n := len(nums)
	points := make([]int, n+2)
	points[0], points[n+1] = 1, 1
	for i := 1; i <= n; i++ {
		points[i] = nums[i-1]
	}

	dp := make([][]int, n+2)
	for d := range dp {
		d = make([]int, n+2)
	}

	for i := n; i >= 0; i-- {
		for j := i + 1; j < n+2; j++ {
			for k := i + 1; k < j; k++ {

			}
		}
	}

}

File: java/src/main/java/cn/az/code/other/dp/BurstingBalloon.java
package cn.az.code.other.dp;

/**
 * @author az
 * @date 2020/5/21
 */
public class BurstingBalloon {

    /**
     *
     * @param nums nums
     * @return coins
     */
    int maxCoins(int[] nums) {
        int len = nums.length;
        int[] points = new int[len + 2];
        points[0] = points[len + 1] = 1;

        for (int i = 1; i <= len; i++) {
            points[i] = nums[i - 1];
        }

        int[][] dp = new int[len + 2][len + 2];

        for (int i = len; i >= 0; i--) {
            for (int j = i + 1; j < len + 2; j++) {
                for (int k = i + 1; k < j; k++) {
                    dp[i][j] = Math.max(dp[i][j], dp[i][k] + dp[k][j] + points[i] * points[k] * points[j]);
                }
            }
        }
        return dp[0][len - 1];
    }
}
